contracts/axelar-gateway/src/test.rs

```diff
@@ -276,3 +276,65 @@ fn rotate_signers() {
         (source_chain, source_address),
     );
 }
+
+#[test]
+fn transfer_operatorship() {
+    let (env, contract_id, client) = setup_env();
+    let operator = Address::generate(&env);
+    let new_operator = Address::generate(&env);
+
+    initialize(&env, &client, operator.clone(), 1, randint(1, 10));
+
+    assert_eq!(client.operator(), operator);
+
+    client
+        .mock_auths(&[MockAuth {
+            address: &operator,
+            invoke: &MockAuthInvoke {
+                contract: &contract_id,
+                fn_name: "transfer_operatorship",
+                args: (&new_operator,).into_val(&env),
```

Comment:
for some methods, we only need to test that the caller is a certain address, so it's a bit unnecessary to have to specify the full invocation everytime. Maybe there's a less brittle way to do this (for e.g. this will break on refactoring, if the function name changes. You could ask in Stellar discord perhaps

Reply:
I agree that for some methods, we don't need to specify the full invocation every time. But for the `transfer_operatorship` unit test, it specifically checks whether the caller is `operator` or not, so I added two cases: (1) only the operator can transfer Operatorship (2) non-operator cannot transfer Operatorship. (Unauthorized). I believe this should break if the function name changes during refactoring. Let me know if you still think we need to remove the mocking of auth for operator in `transfer_operatorship` unit test.

fyi I followed this step to test the sender is authorized: https://discord.com/channels/897514728459468821/1265827786111586344/1292434336712495114

Reply:
Yeah, testing that the caller is the operator is good. I just don't like the dependency on the function name which can break due to refactoring. The actual contract logic doesn't care about the function name. If there's no way to leave the function name, and args unspecified, perhaps we can create a macro that wraps the call to transfer_operatorship and the caller account we provide, and adds the mock auth on top for us. You can investigate as a separate task

Reply:
Got it. I think its worth researching on it. I will investigate as a separate task.